packages/toolpad-app/src/pageViewState.ts

```diff
@@ -21,11 +22,16 @@ function getNodeViewInfo(
     const rect = getRelativeOuterRect(viewElm, elm);
     const props = fiber.child?.memoizedProps ?? {};
 
+    const innerElm = elm.firstElementChild;
```

Comment:
I'm a bit hesitant adding extra dom for measuring the contentBox. Can't we calculate it using the computed properties of `elm`?

Reply:
yeah, i originally thought about it like this but you're right, no need to do it in advance for every element as this rectangle will probably only be used when resizing

Reply:
i need access to the `element` though, so not sure how else to do this, i can give it a try

Reply:
sorry, not sure if i understood your point correctly at first - i added a `div` around each component just to cover the case where people add custom components that could be composed of multiple elements - a div would guarantee that any component inside the container is always wrapped by a single element

if it wasn't for those specific cases, nothing new would need to be added to the dom in this PR

other than that, if i didn't add any new elements to the DOM and just created this new property for each node, would doing things in a different way still be more efficient somehow?

Reply:
> a div would guarantee that any component inside the container is always wrapped by a single element

Shouldn't we add a box container around custom components?

Reply:
> > a div would guarantee that any component inside the container is always wrapped by a single element
> 
> Shouldn't we add a box container around custom components?

We do, but my idea for the inner div was for its bounding box rectangle to be possible to use for resizing the component inside the container later, for example, or to be able to show the margin between the edges of the container and the edges of the component if that becomes customizable later.

Reply:
We can keep this PR on hold until I possibly have to use it, it's not necessary yet anyway.

Reply:
> We do, but my idea for the inner div was for its bounding box rectangle to be possible to use for resizing the component inside the container later, for example, or to be able to show the margin between the edges of the container and the edges of the component if that becomes customizable later.

But that rectangle can also be calculated using a `getRelativeInnerRect` on `elm`, instead of `getRelativeOuterRect` of the child of `elm`

Reply:
> > We do, but my idea for the inner div was for its bounding box rectangle to be possible to use for resizing the component inside the container later, for example, or to be able to show the margin between the edges of the container and the edges of the component if that becomes customizable later.
> 
> But that rectangle can also be calculated using a `getRelativeInnerRect` on `elm`, instead of `getRelativeOuterRect` of the child of `elm`

ok i can give it a try, i was thinking with flexbox as we have it in container it wouldn't be possible, but even if it's not maybe we can switch to a box model with margin and padding instead of flexbox, once we add resizing